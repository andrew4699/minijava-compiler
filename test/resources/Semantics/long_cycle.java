class Main {

  public static void main(String[] args) {
    System.out.println(1);
  }
}

class a extends e {}
class b extends a {}
class c extends b {}
class d extends c {}
class e extends a {}
