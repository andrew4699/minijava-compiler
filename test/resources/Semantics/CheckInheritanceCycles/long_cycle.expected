[GlobalSymbolTable]: Class a is part of an inheritance cycle
[GlobalSymbolTable]: Class b is part of an inheritance cycle
[GlobalSymbolTable]: Class c is part of an inheritance cycle
[GlobalSymbolTable]: Class d is part of an inheritance cycle
[GlobalSymbolTable]: Class e is part of an inheritance cycle
